keps/sig-scheduling/5075-dra-consumable-capacity/README.md

```diff
@@ -0,0 +1,1196 @@
+- Support refined capacity consumption such as predefined blocks and minimum and maximum amount.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+
+Story|Driver|Claim|Context
+---|---|----|---
+1|shared device with capacity|request with resources|network
+2|-|request with resources|gpu
+3|-|request with selector|network
+4|-|request with all resources|network
+5|-|request with requests and limits|network
+6|shared device without capacity|request with no resource|network
+7|-|request with nil resource and AllowShared mode|any
+
+#### Story 1
+A user requests a secondary network based on their bandwidth demands without wanting to specify a device name 
+nor present available bandwidth of each specific device. 
+The DRA scheduler can dynamically select available devices based on their availability.
+
+```yaml
+kind: ResourceSlice
+...
+spec:
+  driver: cni.dra.networking.x-k8s.io
+  devices:
+  - name: eth1
+    basic:
+      shared: "true"
+      attributes:
+        name:
+          string: "eth1"
+      capacity:
+        bandwidth:
+          value: 10Gi
+- name: eth2
+    basic:
+      attributes:
+        name:
+          string: "eth1"
+      capacity:
+        bandwidth:
+          value: 10Gi
+---
+kind: DeviceClass
+metadata:
+  name: vlan-cni.networking.x-k8s.io
+```
+
+With the below ResourceClaim, either `eth1` or `eth2` can be selected. 
+The master field in the CNI config can be filled according to the selected device by the CNI DRA driver.
+
+```yaml
+kind: ResourceClaim
+...
+spec:
+  devices:
+    requests:
+    - name: macvlan
+      deviceClassName: vlan-cni.networking.x-k8s.io
+      resources:
+        requests:
+          bandwidth: "1Gi"
+    config:
+    - requests:
+      - macvlan
+      opaque:
+        driver: cni.dra.networking.x-k8s.io
+        parameters: # CNIParameters with the GVK, interface name and CNI Config (in YAML format).
+          apiVersion: cni.networking.x-k8s.io/v1alpha1
+          kind: CNI
+          ifName: "net1"
+          config:
+            cniVersion: 1.0.0
+            name: net1
+            plugins:
+            - type: macvlan
+              mode: bridge
+              ipam:
+                type: host-local
+                ranges:
+                - - subnet: 10.10.1.0/24
+```
+
+#### Story 2
+A user reserves that fractions of memory for the virtual GPU.
+
+```yaml
+apiVersion: resource.k8s.io/v1beta1
+kind: ResourceClaim
+...
+spec:
+  devices:
+    requests:
+    - name: vgpu0
+      deviceClassName: vgpu.nvidia.com
+      resources:
+        requests:
+          memory: "10Gi"
```

Comment:
How should users think about the requests specified here vs selectors that reference capacity?

Reply:
@mortent 
I think this will be similar to the node selection level where we have node selector to filter (select) the device and we have resource request to request device within the node. In DRA device level, selector is used for filtering the device while resource here is to request for the resource of that device.

Reply:
Yeah, I agree that is probably the most logical way to think about it. But it probably also means that the selector will be evaluated against the full device and not the logical "shared device". It might be a bit surprising as it means that using a selector that is smaller than the request means the claim can never be satisfied, while it also lets users do things like request 10Gb of bandwidth from a device that must have at least 40Gb.